content/asp.net/getting-started/sending-data.md

```diff
@@ -1,20 +1,100 @@
+
+```c#
+public IActionResult Edit(int id, [FromBody] Person person)
+{
+    // ...
+}
+```
+
+Likewise, if you wanted to strictly allow the ID of an object to be set by a routing token, and not overwritten by a later binding operation, you would need to be explicit about that as well. If your incoming URL looked like the following:
+
+    http://www.alpineskihouse.com/peeps/edit/21
+
+...you could then update your signature to include the `[FromRoute]` parameter attribute to have the `id` populated through the route token, and not overwritten by other values.  The end result could end up looking like the following:
+
+```c#
+public IActionResult Edit([FromRoute] int id, [FromBody] Person person)
+{
+    // ...
+}
+```
+
+The default model binder is pretty good at figuring out things like lists as well. For example, we could be asking the user who their closest friends are. Imagine a list of checkboxes on a page that had the following markup:
+
+```html
+<input type="checkbox" name="friend" value="Stephen">Steve
+<input type="checkbox" name="friend" value="Peter">Peter
+<input type="checkbox" name="friend" value="Mark">Mark
+<input type="checkbox" name="friend" value="James">Jimmy
+```
+
+With those fields wrapped in a form element and submitted to an action, we could have the following code in place for the model binder to give us the data:
+
+```c#
+public IActionResult SetClosestFriends(List<string> friends)
+{
+    // ...
+}
+```
+
+In the above example, each friend selected in a checkbox and submitted to the action will be in the `friends` parameter.
+
+Another point to make with the binding process is that you essentially get one opportunity to bind parameter values using any part of the `BODY`. The `BODY` is a stream that can be read from, but you cannot rely on being able to rewind the stream and re-read it. For this reason, a maximum of one parameter may contain a binding attribute that reads from the `BODY` of the request.
 
 ## Next Steps
 
-Give the reader some additional exercises/tasks they can perform to try out what they've just learned.
\ No newline at end of file
+ - Review the [routing](routing.md) concepts to understand how to extract tokens that can be used in the model binding process.
+ - Have a look again at [configuration](configuration.md) or read ahed into [dependency injection](controller-dependencies.md) if you are interested in getting environment- or application-level data into your controller.
+ - Experiment with the sample project to see the different model binding in action. Be sure to set breakpoints so that you can inspect the values of the parameters as your action is entered.
```

Comment:
"see the different model binding in action" -> "see the different kinds of model binding in action"